mender-update/daemon/context.cpp

```diff
@@ -410,10 +410,16 @@ static error::Error UnmarshalJsonStateData(const json::Json &json, StateData &st
 
 	exp_string_vector = json_artifact.Get("PayloadTypes").and_then(json::ToStringVector);
 	SetOrReturnIfError(artifact.payload_types, exp_string_vector);
+	// It's possible for there not to be an initialized update,
+	// if the deployment failed before we could successfully parse the artifact.
+	if (artifact.payload_types.size() == 0) {
+		return error::NoError;
+	}
```

Comment:
Correct.

We could consider making `payload_types` default value something else, to clearly separate it from `zero` payloads, and `not parsed yet` maybe?

Reply:
Yes, or use something else which is even more unambiguously invalid in an artifact. What about `state_data->update_info.id == ""`?

Reply:
I'd rather have an initialised, semantic value tbh

Reply:
Sure, just remember that if you introduce a new value, then it's a database schema update. I think it should be fine though, because all of this happens before the download has started, so there should be no risk of an old client trying to read it.

Reply:
Hmm, didn't think of that.. I'll ponder it a bit

Reply:
Alright, going with `update-info.id` :+1:

Reply:
Decided to use the combination of missing `payload` and `artifact` name as the empty marker